File: javascript/src/Behavioral/Mediator/DvdUpcaseTitle.js
(function() {
  var __hasProp = {}.hasOwnProperty,
    __extends = function(child, parent) { for (var key in parent) { if (__hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; };

  define(['Behavioral/Mediator/DvdTitle'], function(DvdTitle) {
    var DvdUpcaseTitle;
    return DvdUpcaseTitle = (function(_super) {
      __extends(DvdUpcaseTitle, _super);

      DvdUpcaseTitle.prototype.upcaseTitle = null;

      DvdUpcaseTitle.prototype.dvdMediator = null;

      function DvdUpcaseTitle(title, dvdMediator) {
        this.setTitle(title instanceof DvdTitle ? title.getTitle() : title);
        this.resetTitle();
        this.dvdMediator = dvdMediator;
        this.dvdMediator.setUpcase(this);
      }

      DvdUpcaseTitle.prototype.resetTitle = function(title) {
        if (title) {
          this.setTitle(title);
        }
        return this.setUpcaseTitle(this.getTitle().toUpperCase());
      };

      DvdUpcaseTitle.prototype.setSuperTitleUpcase = function() {
        this.setTitle(this.getUpcaseTitle());
        return this.dvdMediator.changeTitle(this);
      };

      DvdUpcaseTitle.prototype.getUpcaseTitle = function() {
        return this.upcaseTitle;
      };

      DvdUpcaseTitle.prototype.setUpcaseTitle = function(upcaseTitle) {
        return this.upcaseTitle = upcaseTitle;
      };

      return DvdUpcaseTitle;

    })(DvdTitle);
  });

}).call(this);

File: go/behavioral/mediator/DvdUpcaseTitle.go
package mediator

import (
	"strings"
)


type DvdUpcaseTitle struct {  
   title string
   upcaseTitle string
   dvdMediator *DvdMediator
}

func NewDvdUpcaseTitle (title string, med *DvdMediator) *DvdUpcaseTitle {
	obj := new(DvdUpcaseTitle)
	obj.title = title
	obj.ResetTitle()
	obj.dvdMediator = med
	med.SetUpcaseTitle(obj)
	return obj
}
    
func NewDvdUpcaseTitleFromObj (dvdTitle *DvdLowercaseTitle, med *DvdMediator) *DvdUpcaseTitle {
    return  NewDvdUpcaseTitle(dvdTitle.GetTitle(), med)
}       

func (this *DvdUpcaseTitle) GetTitle() string {
   return this.title
}

func (this *DvdUpcaseTitle) SetTitle(titleIn string) {
   this.title = titleIn
}

   
func (this *DvdUpcaseTitle) ResetTitle() {
	 this.setUpcaseTitle( strings.ToUpper(this.GetTitle()) )
}

func (this *DvdUpcaseTitle) ResetTitleWithString (title string) {
	 this.SetTitle(title)
	 this.ResetTitle()
}    
   
func (this *DvdUpcaseTitle) SetSuperTitleUpcase() {
	 this.SetTitle(this.GetUpcaseTitle())
	 this.dvdMediator.ChangeUpTitle(this)
}
   
func (this *DvdUpcaseTitle) GetUpcaseTitle() string {
	return this.upcaseTitle
}

func (this *DvdUpcaseTitle) setUpcaseTitle(upcaseTitle string) {
    this.upcaseTitle = upcaseTitle
}
